File: web/src/api/system/dept.ts
import { defHttp } from '@/utils/http';
import { FilterParam } from '/@/utils';
//类型
export interface LoginData {
  username: string;
  password: string;
}
enum Api {
    getList = '/system/dept/getList',
    getParent = '/system/dept/getParent',
    save = '/system/dept/save',
    upStatus = '/system/dept/upStatus',
    del = '/system/dept/del',
    dragDept = '/system/dept/dragDept',
}

//列表数据
export function getList(params: any) {
  params=FilterParam(params)
  return defHttp.get({ url: Api.getList, params:params }, { errorMessageMode: 'message' });
}
//选择数据
export function getParent(params: any) {
  return defHttp.get({ url: Api.getParent, params:params}, { errorMessageMode: 'message' });
}
//提交数据
export function save(params: object) {
    return defHttp.post({ url: Api.save, params:params}, { errorMessageMode: 'message' });
}
//更新状态
export function upStatus(params: object) {
    return defHttp.post({ url: Api.upStatus, params:params}, { errorMessageMode: 'message' });
}
//删除数据
export function del(params: object) {
    return defHttp.delete({ url: Api.del, params:params}, { errorMessageMode: 'message' });
}
//移动数据
export function dragDept(params: object) {
    return defHttp.post({ url: Api.dragDept, params:params}, { errorMessageMode: 'message' });
}

File: api/admin/system/dept.go
package system

import (
	"github.com/suxinwl/GoSuxin/utility/gf"

	"github.com/suxinwl/GoSuxin/framework/frame/g"
)

type DeptListReq struct {
	g.Meta     `path:"/system/dept/getList" tags:"Getlist" method:"get" summary:"获取部门数据"`
	Name       string `p:"name" d:"" dc:"搜索名称"`
	Status     any    `p:"status" d:"" dc:"状态"`
	Createtime string `p:"createtime" d:"" dc:"创建时间"`
}
type DeptListRes struct {
	*gf.R
}

// 获取选项列表
type DeptParentReq struct {
	g.Meta `path:"/system/dept/getParent" noAuth:"1" tags:"getParent" method:"get" summary:"获取选项列表"`
	Id     int64 `p:"id" d:"0" dc:"编辑时自己Id"`
}
type DeptParentRes struct {
	*gf.R
}

// 保存数据
type DeptSaveReq struct {
	g.Meta `path:"/system/dept/save" tags:"Save" method:"post" summary:"保存、编辑部门"`
	Id     int64  `p:"id" d:"0" dc:"数据id，新增时为0"`
	Name   string `p:"name" v:"required#部门名称不能为空" dc:"部门名称"`
	Pid    int    `p:"pid" d:"0" dc:"上级部门id"`
	Remark string `p:"remark" dc:"备注"`
	Status int    `p:"status" d:"0" dc:"状态"`
	Weigh  int    `p:"weigh" d:"0" dc:"排序"`
}
type DeptSaveRes struct {
	*gf.R
}

type DeptStatusReq struct {
	g.Meta `path:"/system/dept/upStatus" tags:"upStatus" method:"post" summary:"更新状态"`
	Id     int64 `p:"id"  v:"required#id不能为空" dc:"更新id"`
	Status uint  `p:"status" v:"required#status不能为空" dc:"状态"`
}
type DeptStatusRes struct {
	*gf.R
}

type DeptDelReq struct {
	g.Meta `path:"/system/dept/del" tags:"Del" method:"delete" summary:"删除"`
	Ids    g.Array `p:"ids"`
}
type DeptDelRes struct {
	*gf.R
}

type DragDeptReq struct {
	g.Meta `path:"/system/dept/dragDept" tags:"dragDept" method:"post" summary:"移动部门"`
	Id     int64 `p:"id" v:"required#id不能为空" dc:"更新id"`
	Pid    int64 `p:"pid" v:"required#pid不能为空" dc:"父级id"`
}
type DragDeptRes struct {
	*gf.R
}
